UI/JustAssembly/ViewModels/ShellViewModel.cs

```diff
@@ -75,8 +83,10 @@ public void CancelCurrentOperation()
 
         public void OpenNewSessionCommandExecuted()
         {
-            NewSessionViewModel newSessionViewModel = new NewSessionViewModel();
+            NewSessionViewModel newSessionViewModel = new NewSessionViewModel(_args);
```

Comment:
This kind of usage of the `NewSessionViewModel` is wrong, because this is the view model for the "New Session" dialog. In this case no dialog is shown and `NewSessionViewModel` should not be used. `AssembliesComparisonViewModel` can be used directly.

Moreover that just code quality - currently there is a big bug. If you start JustAssembly with command-line arguments and click the "Open" button the same arguments are used and the same session is created, i.e. creating new session is impossible.

The bug is caused by the wrong placement of the code for determining if arguments are passed and if the new session dialog should be bypassed. The right location for that is the `OnLoaded` method of the `Shell` class.

Reply:
Alright, let's see if this is better.
I added a new void method `IShellViewModel.OpenNewSessionWithCmdLineArgsCommandExecuted`
This is used in the `Shell` when there's any cmd line args.
I kept the actual validation in `ShellViewModel` because I wanted to show a dialog message to the user when the args are invalid for any reason.

The `ShellViewModel.OpenNewSessionWithCmdLineArgsCommandExecuted` will default to the normal `ShellViewModel.OpenNewSessionCommandExecuted` when the args are invalid (and a message is shown)
If the args are valid, we'll use the `AssembliesComparisonViewModel` directly.

This new method (`OpenNewSessionWithCmdLineArgsCommandExecuted`) is not used as the `ShellViewModel.OpenNewSessionCommand`, so it doesn't intefere with the "Open" button.

If changing the interface is against your policies, let me know and I'll think of an alternative implementation.